src/dos/dos_keyboard_layout.cpp

```diff
@@ -808,7 +808,8 @@ Bitu keyboard_layout::read_codepage_file(const char* codepage_file_name, Bit32s
 		found_at_pos=0x29;
 		size_of_cpxdata=cpi_buf_size;
 	} else {
-		Bit32u dr = (Bit32u)fread(cpi_buf, sizeof(Bit8u), 5, tempfile.get());
+		constexpr auto bytes_to_detect_upx = 5;
+		const auto dr = fread(cpi_buf.data(), sizeof(uint8_t), bytes_to_detect_upx, tempfile.get());
```

Comment:
Maybe `sizeof(cpi_buf[0])` or `sizeof(decltype(cpi_buf)::value_type)`? I've always preferred infer the size from the actual variable where possible.

Same for others below.

Reply:
Yes, the more direct, the better.
I prefer the first (less convoluted syntax), and can store it as a one-time constexpr's that we use throughout. Will be less repetition too; thanks @shermp !

Reply:
added.